pkg/broker/api.go

```diff
@@ -177,82 +177,83 @@ func (b *AwsBroker) Provision(request *osb.ProvisionRequest, c *broker.RequestCo
 	return &response, nil
 }
 
-// Deprovision executed when the osb api receives DELETE /v2/service_instances/:instance_id
-// https://github.com/openservicebrokerapi/servicebroker/blob/v2.13/spec.md#deprovisioning
+// Deprovision is executed when the OSB API receives `DELETE /v2/service_instances/:instance_id`
+// (https://github.com/openservicebrokerapi/servicebroker/blob/v2.13/spec.md#deprovisioning).
 func (b *AwsBroker) Deprovision(request *osb.DeprovisionRequest, c *broker.RequestContext) (*broker.DeprovisionResponse, error) {
-	response := broker.DeprovisionResponse{}
-	si, err := b.db.DataStorePort.GetServiceInstance(request.InstanceID)
-	if err != nil {
-		panic(err)
-	}
-	if si == nil || si.StackID == "" {
-		errmsg := "CloudFormation stackid missing, chances are stack creation failed in an unexpected way, assuming there is nothing to deprovision"
-		glog.Errorln(errmsg)
-		response.Async = false
-		return &response, nil
```

Comment:
Based on [the docs](https://github.com/openservicebrokerapi/servicebroker/blob/v2.13/spec.md#response-6), we now return `410 Gone` instead of `200 OK` if the service instance does not exist.

Also, I removed the empty stack ID handling. Based on how `Provision` works, there shouldn't be a way to create a service instance item with no stack ID.

Reply:
I'm good with this for now, but would want to test on how different platforms respond to the 410, it could result in stuck instances that can't be deleted.